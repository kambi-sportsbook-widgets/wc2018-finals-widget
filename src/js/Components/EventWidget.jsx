import React from 'react'
import PropTypes from 'prop-types'
import { widgetModule, translationModule } from 'kambi-widget-core-library'
import styles from './EventWidget.scss'
import Participants from './Participants'
import BetOffers from './BetOffers'

/**
 * Navigate to event
 * @param {Object} event
 */
const navigateToEvent = event => {
  if (event.event.openForLiveBetting === true) {
    // Navigate to live event
    // http://kambi-sportsbook-widgets.github.io/widget-core-library/module-widgetModule.html#.navigateToLiveEvent__anchor
    widgetModule.navigateToLiveEvent(event.event.id)
  } else {
    // Navigate to prelive event
    // http://kambi-sportsbook-widgets.github.io/widget-core-library/module-widgetModule.html#.navigateToEvent__anchor
    widgetModule.navigateToEvent(event.event.id)
  }
}

/**
 * Check if the event is live and return a label to be added to the title
 * @param event
 * @returns {String} Either the translated label or an empty string
 */
const liveLabel = event => {
  if (event.event.liveBetOffers) {
    // Translate a string
    // http://kambi-sportsbook-widgets.github.io/widget-core-library/module-translationModule.html#.getTranslation__anchor
    return ' - ' + translationModule.getTranslation('liveRightNow')
  }
  return ''
}

class EventWidget extends React.Component {
  /**
   * Called after the component is mounted
   **/
  componentDidMount() {
    this.adaptWidgetHeight()
    window.addEventListener('resize', this.adaptWidgetHeight)
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.adaptWidgetHeight)
  }

  adaptWidgetHeight = () => {
    // Adjust widget height
    // http://kambi-sportsbook-widgets.github.io/widget-core-library/module-widgetModule.html#.adaptWidgetHeight__anchor
    widgetModule.adaptWidgetHeight()
  }

  render() {
    let sliceIndex = this.props.event.betOffers.length > 2 ? 2 : 1
    const betOffersLeft = this.props.event.betOffers.slice(0, sliceIndex)
    const betOffersRight = this.props.event.betOffers.slice(sliceIndex)

    return (
      <div>
        <div className={styles.container}>
          <BetOffers betOffers={betOffersLeft} event={this.props.event.event} />
          <Participants
            event={this.props.event.event}
            flagBaseUrl={this.props.flagBaseUrl}
            iconUrl={this.props.iconUrl}
            onClick={navigateToEvent.bind(null, this.props.event)}
          />
          <BetOffers
            betOffers={betOffersRight}
            event={this.props.event.event}
          />
        </div>
      </div>
    )
  }
}

EventWidget.propTypes = {
  event: PropTypes.object.isRequired,
  flagBaseUrl: PropTypes.string,
  iconUrl: PropTypes.string,
}

export default EventWidget
