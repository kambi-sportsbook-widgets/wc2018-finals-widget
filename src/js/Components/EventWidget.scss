.container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  color: #fff;
  text-shadow: 0 0 2px #000;
  margin: 1em 0;
}

@media screen and (max-width: 768px) {
  .container {
    flex-wrap: wrap;
  }
}
