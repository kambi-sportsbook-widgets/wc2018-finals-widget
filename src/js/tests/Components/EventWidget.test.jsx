import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import EventWidget from '../../Components/EventWidget'

let renderer;

const emptyEvent = {
  betOffers: [],
  event: {},
}

const exampleEvent = {
  "betOffers": [ {}, {}, {}, {} ],
  "event": {
    "countriesEnglishNames": [
      "Country1",
      "Country2",
    ],
    "englishName": "Country1 - Country2",
    "id": 1,
    "start": 1528988400000,
    "state": "NOT_STARTED",
    "type": "ET_MATCH",
  },
}

describe('Event DOM rendering', () => {

  beforeEach(() => {
    renderer = new ReactShallowRenderer()
  })

  it('renders correctly with default props', () => {
    expect(
      renderer.render(
        <EventWidget
          event={emptyEvent}
          flagBaseUrl={null}
          iconUrl={null}
        />
      )
    )
    .toMatchSnapshot()
  })

  it('renders correctly with an example event', () => {
    expect(
      renderer.render(
        <EventWidget
          event={exampleEvent}
          flagBaseUrl={null}
          iconUrl={null}
        />
      )
    )
    .toMatchSnapshot()
  })
})
