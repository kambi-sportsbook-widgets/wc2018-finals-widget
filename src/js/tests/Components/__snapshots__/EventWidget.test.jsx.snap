// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event DOM rendering renders correctly with an example event 1`] = `
<div>
  <div
    className="container"
  >
    <BetOffers
      betOffers={
        Array [
          Object {},
          Object {},
        ]
      }
      event={
        Object {
          "countriesEnglishNames": Array [
            "Country1",
            "Country2",
          ],
          "englishName": "Country1 - Country2",
          "id": 1,
          "start": 1528988400000,
          "state": "NOT_STARTED",
          "type": "ET_MATCH",
        }
      }
    />
    <Participants
      event={
        Object {
          "countriesEnglishNames": Array [
            "Country1",
            "Country2",
          ],
          "englishName": "Country1 - Country2",
          "id": 1,
          "start": 1528988400000,
          "state": "NOT_STARTED",
          "type": "ET_MATCH",
        }
      }
      flagBaseUrl={null}
      iconUrl={null}
      onClick={[Function]}
    />
    <BetOffers
      betOffers={
        Array [
          Object {},
          Object {},
        ]
      }
      event={
        Object {
          "countriesEnglishNames": Array [
            "Country1",
            "Country2",
          ],
          "englishName": "Country1 - Country2",
          "id": 1,
          "start": 1528988400000,
          "state": "NOT_STARTED",
          "type": "ET_MATCH",
        }
      }
    />
  </div>
</div>
`;

exports[`Event DOM rendering renders correctly with default props 1`] = `
<div>
  <div
    className="container"
  >
    <BetOffers
      betOffers={Array []}
      event={Object {}}
    />
    <Participants
      event={Object {}}
      flagBaseUrl={null}
      iconUrl={null}
      onClick={[Function]}
    />
    <BetOffers
      betOffers={Array []}
      event={Object {}}
    />
  </div>
</div>
`;
